pkg/store/bucket.go

```diff
@@ -661,10 +700,17 @@ func debugFoundBlockSetOverview(logger log.Logger, mint, maxt int64, lset labels
 }
 
 // Series implements the storepb.StoreServer interface.
-// TODO(bwplotka): It buffers all chunks in memory and only then streams to client.
-// 1. Either count chunk sizes and error out too big query.
-// 2. Stream posting -> series -> chunk all together.
 func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_SeriesServer) error {
+	{
+		span, _ := tracing.StartSpan(srv.Context(), "bucket_store_gate")
```

Comment:
If I'd be troubleshooting this, I'd be more interested in `queryGate.IsMyTurn` instead of `bucket_store_gate`

Reply:
Are you referring to the name itself? It seems like the convention here is to use underscore between different parts so perhaps `store_query_gate_ismyturn` would be more fitting. What do you think, @rafaeljesus ?

Reply:
true let's stick with the current convention, makes total sense, also `store_query_gate_ismyturn` LGTM